go-controller/pkg/ovn/egressip.go

```diff
@@ -1516,23 +1537,44 @@ func (e *egressIPZoneController) deletePodEgressIPAssignment(egressIPName string
 			metrics.RecordEgressIPUnassign(duration)
 		}()
 	}
-
+	isLocalZoneEgressNode, loadedEgressNode := e.nodeZoneState.Load(status.Node)
 	isLocalZonePod, loadedPodNode := e.nodeZoneState.Load(pod.Spec.NodeName)
+	var nextHopIP string
+	// node may not exist - attempt to retrieve it
+	eNode, err := e.watchFactory.GetNode(status.Node)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return fmt.Errorf("failed to add pod %s/%s because failed to lookup node %s: %v", pod.Namespace, pod.Name,
+			pod.Spec.NodeName, err)
+	}
+	if err == nil {
+		eIPIP := net.ParseIP(status.EgressIP)
+		isOVNManagedNetwork, err := util.IsOVNManagedNetwork(eNode, eIPIP)
+		if err != nil {
+			return fmt.Errorf("failed to delete egress IP %s (%s) because failed to determine if EgressIP is OVN managed: %v",
+				egressIPName, status.EgressIP, err)
+		}
+		nextHopIP, err = e.getNextHop(status.Node, status.EgressIP, egressIPName, isLocalZoneEgressNode, isOVNManagedNetwork)
+		if err != nil {
+			return fmt.Errorf("failed to delete egress IP %s (%s) because unable to determine next hop: %v", egressIPName,
```

Comment:
wait it looks like you still try to delete the policy anyway even if you dont have nextHopIP. So why are you returning error here?

```
func (e *egressIPZoneController) deleteReroutePolicyOps(ops []ovsdb.Operation, podIPNets []*net.IPNet, status egressipv1.EgressIPStatusItem, egressIPName, nextHopIP string) ([]ovsdb.Operation, error) {
	isEgressIPv6 := utilnet.IsIPv6String(status.EgressIP)
	var err error
	// Handle all pod IPs that match the egress IP address family
	for _, podIPNet := range util.MatchAllIPNetFamily(isEgressIPv6, podIPNets) {
		filterOption := fmt.Sprintf("%s.src == %s", ipFamilyName(isEgressIPv6), podIPNet.IP.String())
		p := func(item *nbdb.LogicalRouterPolicy) bool {
			return item.Match == filterOption && item.Priority == types.EgressIPReroutePriority && item.ExternalIDs["name"] == egressIPName
		}
		if nextHopIP != "" {
			ops, err = libovsdbops.DeleteNextHopFromLogicalRouterPoliciesWithPredicateOps(e.nbClient, ops, types.OVNClusterRouter, p, nextHopIP)
			if err != nil {
				return nil, fmt.Errorf("error removing nexthop IP %s from egress ip %s policies on router %s: %v",
					nextHopIP, egressIPName, types.OVNClusterRouter, err)
			}
		} else {
			klog.Errorf("Caller failed to pass next hop for EgressIP %s and IP %s. Deleting all LRPs. This will break HA momentarily",
				egressIPName, status.EgressIP)
			// since next hop was not found, delete everything to ensure no stale entries however this will break load
			// balancing between hops, but we offer no guarantees except one of the EIPs will work
			ops, err = libovsdbops.DeleteLogicalRouterPolicyWithPredicateOps(e.nbClient, ops, types.OVNClusterRouter, p)
			if err != nil {
				return nil, fmt.Errorf("failed to create logical router policy operations on ovn_cluster_router: %v", err)
			}
		}
	}
	return ops, nil
}
```

Reply:
Fixed. We don't return error just add a warning level log.